File: src/main/java/jisop/command_line/parse/MergedParsedArguments.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package jisop.command_line.parse;

import jisop.infrastructure.Lists;

import java.util.stream.Stream;

/**
 *
 * @author mathieu
 */
class MergedParsedArguments extends ParsedArguments {
    private final ParsedArguments _first;
    private final ParsedArguments _second;

    public MergedParsedArguments(ParsedArguments first, ParsedArguments second){
        super(
                Lists.union(first.argumentWithOptions, second.argumentWithOptions),
                Lists.union(first.recognizedArguments, second.recognizedArguments),
                Lists.intersect(first.unRecognizedArguments, second.unRecognizedArguments)
                );
         _first = first;
        _second = second;

    }
    @Override
    public Stream<String> invoke() {
        Stream<String> r=Stream.empty();
        r = Stream.concat(r, _first.invoke());
        r = Stream.concat(r, _second.invoke());
        return r;
    }
}

File: src/main/java/isop/command_line/parse/MergedParsedArguments.kt
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package isop.command_line.parse

import isop.infrastructure.Lists

import java.util.stream.Stream

/**
 *
 * @author mathieu
 */
internal class MergedParsedArguments(private val _first: ParsedArguments, private val _second: ParsedArguments) : ParsedArguments(Lists.union(_first.argumentWithOptions, _second.argumentWithOptions), Lists.union(_first.recognizedArguments, _second.recognizedArguments), Lists.intersect(_first.unRecognizedArguments, _second.unRecognizedArguments)) {
    override fun invoke(): Stream<String> {
        var r = Stream.empty<String>()
        r = Stream.concat(r, _first.invoke())
        r = Stream.concat(r, _second.invoke())
        return r
    }
}
